docs/cicd_notes.md

```diff
@@ -0,0 +1,7 @@
+# CI/CD notes
+## Build and push test container GH Action
+We developed a [GitHub Action](.github/workflows/build_container.yaml) to build the test container and push it to Docker Hub when the files it depends on are changed on a push to main or a pull request to main. To do it, we followed the documentation of the [checkout action](https://github.com/marketplace/actions/checkout), [build and push action](https://github.com/marketplace/actions/build-and-push-docker-images), and [Docker metadata action](https://github.com/marketplace/actions/docker-metadata-action).  
+
+The developed action will build and push to Docker Hub the test container every time the files it depends on have changed either on the main branch or in a pull request. Also, to help differentiate between these changes, the uploaded image will have a **different tag** depending on if it was built after pushing to main -it will have the **main** tag- or if it was built for a pull request -where it will have the **pr-<PR number>** tag. This will help to know which image version the tests must be run against.
```

Comment:
It does not look like it works that way. It simply uses the same tags as there were already in the repo, not different ones depending on the event.
At any rate, it does not make any sense to create images on pull request. First because you simply can't if it's done by a different user (GitHub actions sandbox). Second, because if it's you you're simply pushing to a different branch, something you're *excluding* in your rules.
Not the main objective of this, and if it works, it works, but you should have checked out how GHAs work a bit more extensively.